lib/rucio/core/rse.py

```diff
@@ -549,14 +557,15 @@ def get_rses_with_attribute_value(key, value, lookup_key, session=None):
                        .join(models.RSE, models.RSE.id == models.RSEAttrAssociation.rse_id)\
                        .join(subquery, models.RSEAttrAssociation.rse_id == subquery.c.rse_id)\
                        .filter(models.RSE.deleted == false(),
-                               models.RSEAttrAssociation.key == lookup_key)
+                               models.RSEAttrAssociation.key == lookup_key)\
+                       .filter(models.RSE.vo == vo)
```

Comment:
This VO filter should be added to the `.filter` call above rather than having its own.